src/components/profile-sidebar/ProfileSidebar.jsx

```diff
@@ -1,54 +1,47 @@
-import React, { useMemo, useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import { Tabs, Box, Tab } from "@mui/material";
 import { RightChevron } from "../icons";
 import styles from "./ProfileSidebar.module.css";
 import { getSelectedColor } from "./components/tab-content-styles";
-import PropTypes from 'prop-types';
-function TabPanel(props) {
-  const { children, value, index, ...other } = props;
-  return (
-    <div
-      role="tabpanel"
-      hidden={value !== index}
-      id={`vertical-tabpanel-${index}`}
-      aria-labelledby={`vertical-tab-${index}`}
-      {...other}
-    >
-      {value === index && (
-        <Box sx={{ p: 1 }}>
-          {children}
-        </Box>
-      )}
-    </div>
-  );
-}
+import { useLocation, useNavigate } from "react-router-dom";
 
-TabPanel.propTypes = {
-  children: PropTypes.node,
-  index: PropTypes.number.isRequired,
-  value: PropTypes.number.isRequired,
-};
-
-function getTabIndexProps(index) {
-  return {
-    id: `vertical-tab-${index}`,
-    'aria-controls': `vertical-tabpanel-${index}`,
-  };
-}
-
-export function ProfileSidebar({ SidebarOptions, titleSetter }) {
+export function ProfileSidebar({
+  SidebarOptions,
+  titleSetter,
+  setselectedOrderId,
+}) {
   const [value, setValue] = useState(0);
+  const location = useLocation();
+  const navigate = useNavigate();
 
   const handleChange = (event, newValue) => {
     setValue(newValue);
     titleSetter(newValue);
+    setselectedOrderId("");
+    const newPath =
+      newValue !== 0
+        ? SidebarOptions[newValue].label.toLowerCase().split(" ").join("-")
```

Comment:
should have this whole operation in a clearly named variable, or be it's own function